spec/controllers/allocation_staff_controller_spec.rb

```diff
@@ -28,9 +28,22 @@
     end
 
     describe '#index' do
-      context 'with previous allocations for this POM' do
-        let(:alice) { poms.first }
+      let(:alice) { poms.first }
+
+      context 'with an allocation from a previous prison' do
+        before do
+          create(:case_information, offender: build(:offender, nomis_offender_id: offender_no))
+          create(:allocation_history, prison: build(:prison).code, nomis_offender_id: offender_no)
+        end
```

Comment:
This `prison: build(:prison).code` inside allocation history, is a different prison to the `prison_id: prison_code` passed to the index action? Which is how this test is able to reproduce the bug?

Reply:
Yes